pkg/network/tracer/connection/ebpfless/tcp_processor.go

```diff
@@ -61,13 +61,17 @@ func (st *connectionState) hasMissedHandshake() bool {
 
 // TCPProcessor encapsulates TCP state tracking for the ebpfless tracer
 type TCPProcessor struct {
-	conns map[network.ConnectionTuple]connectionState
+	// pendingConns contains connections with tcpState == connStatAttempted
+	pendingConns map[network.ConnectionTuple]*connectionState
+	// establishedConns contains connections with tcpState == connStatEstablished
+	establishedConns map[network.ConnectionTuple]*connectionState
```

Comment:
The `connectionState` is also turned into a pointer to make moving it between maps easier (it probably should have been a pointer from the start tbh)